tidy3d/plugins/adjoint/components/data/data_array.py

```diff
@@ -312,28 +313,37 @@ def isel(self, **isel_kwargs) -> JaxDataArray:
 
         return self_sel
 
-    def sel(self, indexers: dict = None, method: str = "nearest", **sel_kwargs) -> JaxDataArray:
+    def sel(self, indexers: dict = None, method: str | None = None, **sel_kwargs) -> JaxDataArray:
         """Select a value from the :class:`.JaxDataArray` by indexing into coordinate values."""
+        if method is None:
+            return self._sel_exact(sel_kwargs)
+        elif method == "nearest":
+            return self._sel_nearest(sel_kwargs)
+        else:
+            raise NotImplementedError(f"Unkown selection method: {method}.")
+
+    def _sel_exact(self, sel_kwargs: dict) -> JaxDataArray:
         isel_kwargs = {}
-        for coord_name, sel_kwarg in sel_kwargs.items():
-            coord_list = self.get_coord_list(coord_name)
-            if isinstance(sel_kwarg, (tuple, list, np.ndarray)):
-                sel_kwarg = list(sel_kwarg)
-                isel_kwargs[coord_name] = []
-                for _sel_kwarg in sel_kwarg:
-                    if _sel_kwarg not in coord_list:
-                        raise DataError(
-                            f"Could not select '{coord_name}={_sel_kwarg}', value not found."
-                        )
-                    coord_index = coord_list.index(_sel_kwarg)
-                    isel_kwargs[coord_name].append(coord_index)
-            else:
-                if sel_kwarg not in coord_list:
-                    raise DataError(
-                        f"Could not select '{coord_name}={sel_kwarg}', value not found."
-                    )
-                coord_index = coord_list.index(sel_kwarg)
-                isel_kwargs[coord_name] = coord_index
+        for coord_name, vals in sel_kwargs.items():
+            coord_list = np.asarray(self.get_coord_list(coord_name))
+            vals = np.atleast_1d(vals)
+            dist = np.abs(coord_list[None] - vals[:, None])
+            indices = np.where(np.isclose(dist, 0))[1]
+            if indices.size == 0:
+                raise DataError(
+                    f"Could not select '{coord_name}={vals}', some values were not found."
+                )
+            isel_kwargs[coord_name] = np.squeeze(indices)
+        return self.isel(**isel_kwargs)
+
+    def _sel_nearest(self, sel_kwargs: dict) -> JaxDataArray:
+        isel_kwargs = {}
+        for coord_name, vals in sel_kwargs.items():
+            coord_list = np.asarray(self.get_coord_list(coord_name))
+            vals = np.asarray(vals)
+            dist = np.abs(coord_list[:, None] - vals[None])
```

Comment:
can you explain the indexing here? so it seems to me that we add a dimension to `vals` so it would be shaped `(N,) -> (1, N)` and then `coord_list` would be `(M,) -> (M, 1)`. And then we have a distance `(M, N)`. We take argmin with respect to the `coord_list` axis, giving `(N,)` values corresponding to the smallest distance between each value to all of the coords?

Reply:
Yes exactly, which is then an index array of shape `(M,)` that can be used to index into `coord_list`. I came across a bug though, in that this does not handle non-numeric types such as coords of the form `["+", "-"]`. Have to come up with a fix for that.

Reply:
ah.. maybe in that case we should just revert to the exact selection? not sure how xarray handles this.